nixos/modules/installer/tools/nixos-rebuild.sh

```diff
@@ -422,13 +422,22 @@ if [ -z "$rollback" ]; then
         fi
         copyToTarget "$pathToConfig"
         targetHostCmd nix-env -p "$profile" --set "$pathToConfig"
-    elif [ "$action" = test -o "$action" = build -o "$action" = dry-build -o "$action" = dry-activate ]; then
+    elif [ "$action" = build ]; then
         if [[ -z $flake ]]; then
             pathToConfig="$(nixBuild '<nixpkgs/nixos>' -A system -k "${extraBuildFlags[@]}")"
         else
             nix build "$flake#$flakeAttr.config.system.build.toplevel" "${extraBuildFlags[@]}" "${lockFlags[@]}"
             pathToConfig="$(readlink -f ./result)"
         fi
+    elif [ "$action" = test -o "$action" = dry-build -o "$action" = dry-activate ]; then
+        if [[ -z $flake ]]; then
+            pathToConfig="$(nixBuild '<nixpkgs/nixos>' --no-out-link -A system -k "${extraBuildFlags[@]}")"
```

Comment:
Wouldn't it be easier to just add `--no-out-link` to `extraBuildFlags`?

Reply:
The problem is that the flake-specific part is structured differently in the `build` and the `test` case. The nix-command equivalent to `--no-out-link` seems to be called `--no-link`, but the code that's already there (L418-421) didn't use it and I assumed that maybe there's a reason why.